sdl-core/src/main/scala/io/smartdatalake/workflow/dataframe/GenericDataFrame.scala

```diff
@@ -253,6 +253,9 @@ trait GenericSchema extends GenericTypedObject {
   def sql: String
   def add(colName: String, dataType: GenericDataType): GenericSchema
   def add(field: GenericField): GenericSchema
+  def addIfNotExists(colName: String, dataType: GenericDataType): GenericSchema = {
+    if (!columnExists(colName)) add(colName, dataType) else this
```

Comment:
We could log a info message here if the column is automatically added. Otherwise it's a bit hidden and hard to trace.

Reply:
yes, done.